x-pack/solutions/observability/plugins/observability/public/pages/alert_details/components/case_links.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiFlexGroup, EuiFlexItem, EuiLink, EuiText } from '@elastic/eui';
+import { css } from '@emotion/react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { ALERT_CASE_IDS } from '@kbn/rule-data-utils';
+import { useFetchBulkCases } from '../../../hooks/use_fetch_bulk_cases';
+import { useCaseLinks } from '../hooks/use_case_links';
+import { type TopAlert } from '../../../typings/alerts';
+
+const caseLinkStyle = css`
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  max-width: 150px;
+`;
+
+interface Props {
+  alert: TopAlert | null;
+}
+
+export function CaseLinks({ alert }: Props) {
+  const { cases } = useFetchBulkCases({ ids: alert?.fields[ALERT_CASE_IDS] || [] });
+  const { firstCaseLink, casesOverviewLink } = useCaseLinks(cases);
+  if (!cases?.length) return null;
+  return (
+    <EuiFlexItem grow={false}>
+      <EuiFlexGroup gutterSize="xs" alignItems="center">
+        <EuiFlexItem grow={false}>
+          <EuiText size="s" color="subdued">
+            <FormattedMessage
+              id="xpack.observability.pages.alertDetails.cases.title"
+              defaultMessage="{numCases, plural, =1 {Case} other {Cases}}:"
+              values={{ numCases: cases.length }}
+            />
+          </EuiText>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          {!!firstCaseLink ? (
+            <EuiLink
+              data-test-subj="xpack.observability.pages.alertDetails.cases.caseLink"
+              href={firstCaseLink}
+              css={caseLinkStyle}
+              target="_blank"
```

Comment:
Do we want to open in a new tab?

Reply:
https://github.com/elastic/kibana/pull/222903/commits/f13615833e85e1f13b4539f70e9afe7ab34b9a2a